Pulse.Core/EncryptedConnection.cs

```diff
@@ -0,0 +1,63 @@
+using System.Threading.Channels;
+using Pulse.Core.Connections;
+
+namespace Pulse.Core;
+
+internal class EncryptedConnection : IConnection
+{
+    private readonly IConnection actualConnection;
+    private readonly PacketEncryptor encryptor;
+    public ChannelReader<Packet> IncomingAudio { get; }
+
+    public EncryptedConnection(IConnection actualConnection, PacketEncryptor encryptor)
+    {
+        if (!encryptor.Ready)
+        {
+            throw new InvalidOperationException("Encryptor is not properly configured!");
```

Comment:
What exactly is misconfigured?
Use a more helpful error message

Reply:
There are only two options so I left it like that for readability, using `Ready` is much more readable.